Src/FluentAssertions/Types/TypeAssertions.cs

```diff
@@ -644,8 +633,14 @@ public AndConstraint<TypeAssertions> NotBeDerivedFrom<TBaseClass>(string because
         /// <param name="becauseArgs">
         /// Zero or more objects to format using the placeholders in <paramref name="because"/>.
         /// </param>
+        /// <exception cref="InvalidOperationException"><see cref="ReferenceTypeAssertions{Type, TypeAssertions}.Subject"/> is not a class.</exception>
         public AndConstraint<TypeAssertions> BeSealed(string because = "", params object[] becauseArgs)
         {
+            Execute.Assertion
```

Comment:
Ah, will need some addition work there.

Reply:
Would you rather see
```csharp
if (Subject is null)
{
    Execute.Assertion
        .BecauseOf(because, becauseArgs)
        .ForCondition(false)
        .FailWith(
            $"Expected {{context:type}} to explicitly implement {interfaceType.FullName}.{name}" +
            $"({GetParameterString(parameterTypes)}){{reason}}, but {{context:type}} is <null>.");

    return new AndConstraint<TypeAssertions>(this);
}
```

or

```csharp
bool continueAsserting = Execute.Assertion
    .BecauseOf(because, becauseArgs)
    .ForCondition(Subject is not null)
    .FailWith(
        $"Expected {{context:type}} to explicitly implement {interfaceType.FullName}.{name}" +
        $"({GetParameterString(parameterTypes)}){{reason}}, but {{context:type}} is <null>.");

if (continueAsserting)
{
    Subject.Should().Implement(interfaceType, because, becauseArgs);

    var explicitlyImplementsMethod = Subject.HasMethod($"{interfaceType.FullName}.{name}", parameterTypes);

    Execute.Assertion
        .BecauseOf(because, becauseArgs)
        .ForCondition(explicitlyImplementsMethod)
        .FailWith(
            $"Expected {Subject.FullName} to explicitly implement {interfaceType.FullName}.{name}" +
            $"({GetParameterString(parameterTypes)}){{reason}}, but it does not.");
}
```

or something else?

Reply:
I usually use the second example.

Reply:
Applied